AutorestSwiftTest/AutoRestResourceFlatteningTest.swift

```diff
@@ -44,21 +44,304 @@ class AutoRestResourceFlatteningTest: XCTestCase {
+                let details = self.errorDetails(for: error, withResponse: httpResponse)
+                print("test failed. error=\(details)")
+                XCTFail("Call autorestresourceflatteningtestservice.putArray failed")
+            }
+        }
+        wait(for: [expectation], timeout: 5.0)
+    }
+
+    func test_resourceFlattening_getDictionary() throws {
+        let expectation =
+            XCTestExpectation(description: "Call autorestresourceflatteningtestservice.getDictionary succeed")
+        client.autorestresourceflatteningtestservice.getDictionary { result, httpResponse in
+            switch result {
+            case .success:
+                expectation.fulfill()
+            case let .failure(error):
+                let details = self.errorDetails(for: error, withResponse: httpResponse)
+                print("test failed. error=\(details)")
+                XCTFail("Call autorestresourceflatteningtestservice.getDictionary failed")
+            }
+        }
+        wait(for: [expectation], timeout: 5.0)
+    }
+
+    func test_resourceFlattening_putDictionary() throws {
+        let expectation =
+            XCTestExpectation(
+                description: "Call autorestresourceflatteningtestservice.putDictionary succeed"
+            )
+        let dictionary = [
+            "Resource1": FlattenedProduct(
+                pName: nil,
+                typePropertiesType: nil,
+                provisioningStateValues: nil,
+                provisioningState: nil,
+                id: nil,
+                type: "Flat",
+                tags: ["tag1": "value1", "tag2": "value3"],
+                location: "West US",
+                name: "Product1"
+            ),
+            "Resource2": FlattenedProduct(
+                pName: nil,
+                typePropertiesType: nil,
+                provisioningStateValues: nil,
+                provisioningState: nil,
+                id: nil,
+                type: "Flat",
+                tags: nil,
+                location: "Building 44",
+                name: "Product2"
+            )
+        ]
+        client.autorestresourceflatteningtestservice.put(dictionary: dictionary) { result, httpResponse in
+            switch result {
+            case .success:
+                expectation.fulfill()
+            case let .failure(error):
+                let details = self.errorDetails(for: error, withResponse: httpResponse)
+                print("test failed. error=\(details)")
+                XCTFail("Call autorestresourceflatteningtestservice.putDictionary failed")
+            }
+        }
+        wait(for: [expectation], timeout: 5.0)
+    }
+
+    func test_resourceFlattening_getResourceCollection() throws {
+        let expectation =
+            XCTestExpectation(description: "Call autorestresourceflatteningtestservice.getResourceCollection succeed")
+        client.autorestresourceflatteningtestservice.getResourceCollection { result, httpResponse in
```

Comment:
@sacheu is there a more concise way of getting to the methods? This seems very verbose, but may just be a result of the swagger.

Reply:
I have been thinking about that too. How to make the test method more concise. Compare to other languages, swift test cases are more verbose.  I think part of the problem is this pattern `            switch result {
            case .success:
                expectation.fulfill()
            case let .failure(error):
             XCTFail("some error string")
`

Reply:
Part of ^ being verbose because that makes debugger earlier. I can set breakpoints in success and failure cases and inspect the httpresponse when things failed.

Reply:
I think switching on the result is fine. My question is with the `client.omgwhatasuperlongvariablenamethisis.getResourceCollection`. Is this simply a result of the swagger using an operation group?  It would seem `client.getResourceCollection` would be preferrable.

Reply:
Yes. that is because of operation group. Per our discussion in our meeting earlier, we will separate operations even if there is only 1 operation group (as long as that operation group has a name).